### PR TITLE
2nd attempt and typehinting for sessions

### PR DESCRIPTION
**Pull request by Simbiat**
https://github.com/JetBrains/phpstorm-stubs/pull/1140 failed, but as per suggestion in https://youtrack.jetbrains.com/issue/WI-60588 retrying the same with removal of `@since 5.4` as well.

### CODE REVIEW

File: session/session.php [hunk 1]
```diff
@@ -174,7 +174,7 @@ function session_start(array $options = []): bool {}
  * If it is used without active session, it omits collision check.
  * @since 7.1
  */
-function session_create_id(string $prefix): string|false {}
+function session_create_id(string $prefix = ""): string|false {}
```
Comment (williamdes) [inline session/session.php:177]:
```suggestion
function session_create_id(string $prefix = ''): string|false {}
```

File: session/session.php [hunk 2]
```diff
@@ -238,26 +238,27 @@ function session_unset() {}
  * The garbage collector, this is executed when the session garbage collector
  * is executed and takes the max session lifetime as its only parameter.
  * </p>
- * @param callable $create_sid [optional]
+ * @param ?callable $create_sid [optional]
```
Comment (williamdes) [inline session/session.php:241]:
```suggestion
 * @param callable|null $create_sid [optional]
```

File: session/session.php [hunk 3]
```diff
@@ -238,26 +238,27 @@ function session_unset() {}
...
  * </p>
- * @param callable $create_sid [optional]
+ * @param callable|null $create_sid [optional]
  * <p>This callback is executed when a new session ID is required.
  * No parameters are provided, and the return value should be a string that is a valid
  * session ID for your handler.</p>
- * @param callable $validate_sid [optional]
- * @param callable $update_timestamp [optional]
+ * @param ?callable $validate_sid [optional]
+ * @param ?callable $update_timestamp [optional]
```
Comment (williamdes) [inline session/session.php:246]:
```suggestion
 * @param callable|null $validate_sid [optional]
 * @param callable|null $update_timestamp [optional]
```

File: session/SessionHandler.php [hunk 4]
```diff
@@ -29,9 +28,8 @@ public function close();
      * The return value (usually TRUE on success, FALSE on failure).
      * Note this value is returned internally to PHP for processing.
      * </p>
-     * @since 5.4
```
Comment (williamdes) [inline session/SessionHandler.php:32]: I am not to much fan of removing the `since` tags, it always helps building old compat code

**Review by williamdes — APPROVED** [review]

**Comment by Simbiat** [comment]
@williamdes, I hear you, but this was suggested in the ticket as potential solution. Code inspector now shows the same 2 errors, which we can't avoid, so need to wait for @DmitryTronin or someone else from his team to look at this closer.

**Comment by DmitryTronin** [comment]
@Simbiat 
Sorry for the lack of feedback on this.
We took a look at this & failing tests once again and came up with another suggestion: to use `LanguageLevelTypeAware` in parameters definition. Here's an example of how you can use them: https://gist.github.com/DmitryTronin/3bdb3967b5fc6d1b70fd448ffc9dde40
I've used your revision 7d6658592f83324bd979628cdacefd623b175864 named "Removed functions from SessionHandler" to create this example. I've run our tests against the "fixed" SessionHandler class above & it went without failures.

A quick rundown on required changes:
- there shouldn't be return type hints (as explained [here](https://github.com/JetBrains/phpstorm-stubs/pull/1140#issuecomment-853065698)).
- `@since 5.4 tag` should be present, my assumption on [YouTrack](https://youtrack.jetbrains.com/issue/WI-60588#focus=Comments-27-4964766.0-0) was incorrect - if `@since` tag is missing, we assume there's a `@since 5.3` tag. That's an undocumented behavior.
- parameter type hints should be written using `#[LanguageLevelTypeAware()]` construcion.

**Comment by Simbiat** [comment]
Restored `@since` and added `LanguageLevelTypeAware`. Not sure if correctly, though.

**Comment by DmitryTronin** [comment]
We've adjusted the changes you proposed and pushed in https://github.com/JetBrains/phpstorm-stubs/pull/1190. Everything should be looking fine now.
I've also checked https://youtrack.jetbrains.com/issue/WI-60588 - the example shared there works fine now on the latest stubs version.

**Pull request closed without merging**